torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -964,6 +985,25 @@ def test_all_reduce_sum_cuda(self):
                 rank_to_GPU,
             )
 
+
+        @require_backends_available({"gloo", "nccl"})
+        @skip_if_no_gpu
+        def test_all_reduce_sum_cuda_async(self):
+            group, group_id, rank = self._init_global_test()
+            rank_to_GPU = self._init_multigpu_helper()
+            self._test_all_reduce_helper(
+                group,
+                group_id,
```

Comment:
Would be nice to add tests for all the collectives we have enabled to ensure they're profiled correctly.